STEP.grammar

```diff
@@ -116,42 +116,47 @@ Productions
   loop_stmt = loop_stmts nl;
   loop_stmts = {one} loopifstmt | {two} whilestmt | {three} forstmt | {four} vardcl | {five} assstmt | {six} funccall | {seven} retstmt | ;
   loopifbody = {one} loop_stmt | {two} continue nl | {three} break nl;
-  ifstmt = {nonelse} if lparen cond rparen stmt* endif| {withelse} if lparen cond rparen [fst]:stmt* else [snd]:stmt* endif;
-  loopifstmt = {noelse} if lparen cond rparen loopifbody* endif | {withelse} if lparen cond rparen [true]:loopifbody* else [false]:loopifbody* endif;
-  whilestmt = repeatwhile lparen cond rparen loop_stmt* endwhile;
+  ifstmt = {nonelse} if lparen logicexpr rparen stmt* endif| {withelse} if lparen logicexpr rparen [fst]:stmt* else [snd]:stmt* endif;
+  loopifstmt = {noelse} if lparen logicexpr rparen loopifbody* endif | {withelse} if lparen logicexpr rparen [true]:loopifbody* else [false]:loopifbody* endif;
+  whilestmt = repeatwhile lparen logicexpr rparen loop_stmt* endwhile;
   forstmt = repeatfor lparen for_iter_opt to [fst]:expr comma increase_decrease [snd]:expr rparen loop_stmt* endfor;
   for_iter_opt = {one} numdcl | {two} assstmt | {three} id arrindex?;
   increase_decrease = {one} increaseby | {two} decreaseby;
   // switchstmt = switch lparen id arrindex? rparen [fst]:nl [snd]:nl* whencase* otherwisecase? endswitch;
-  whencase = when numliteral do nl stmt* fallthrough_stmt?;
-  fallthrough_stmt = fallthrough [fst]:nl [snd]:nl*;
-  otherwisecase = otherwisedo nl stmt*;
-  assstmt = id arrindex? assign expr;
+  // whencase = when numliteral do nl stmt* fallthrough_stmt?;
+  // fallthrough_stmt = fallthrough [fst]:nl [snd]:nl*;
+  // otherwisecase = otherwisedo nl stmt*;
```

Comment:
Skal de fjernes? Vi har jo git history, hvis vi får brug for dem.

Reply:
i got u